spanner/spanner_snippets/spanner/spanner_read_only_transaction_with_array_of_proto_column.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package spanner
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"cloud.google.com/go/spanner"
+	pb "github.com/GoogleCloudPlatform/golang-samples/spanner/spanner_snippets/spanner/testdata/protos"
+	"google.golang.org/api/iterator"
+	"google.golang.org/api/option"
+)
+
+// [START spanner_read_only_transaction_with_array_of_proto_column]
+
+func readOnlyTransactionWithArrayOfProtoMsgAndEnum(w io.Writer, db string) error {
+	ctx := context.Background()
+	endpoint := "staging-wrenchworks.sandbox.googleapis.com:443"
+	options := []option.ClientOption{option.WithEndpoint(endpoint)}
+	client, err := spanner.NewClient(ctx, db, options...)
+	if err != nil {
+		return err
+	}
+	defer client.Close()
+
+	ro := client.ReadOnlyTransaction()
+	defer ro.Close()
+	stmt := spanner.Statement{SQL: `SELECT SingerId, FirstName, LastName, SingerInfo, SingerGenre FROM Singers`}
+	iter := ro.Query(ctx, stmt)
+	defer iter.Stop()
```

Comment:
There are two calls to `iter.Stop()` here. I *think* you only need one, even though you're assigning another instance further on.

Reply:
The general best practice is to stop the iterator after using it.
Here we need to use `iter.Stop()` at both places because the `RowIterator` object returned by `ro.Query()` and `ro.Read()` are different and we need to close both the iterator objects that are returned.